pyjelly/serialize/encode.py

```diff
@@ -87,39 +80,100 @@ def encode_iri(self, iri_string: str) -> RowsAnd[jelly.RdfIri]:
 
         prefix_index = self.prefixes.encode_prefix_term_index(prefix)
         name_index = self.names.encode_name_term_index(name)
-        return term_rows, jelly.RdfIri(prefix_id=prefix_index, name_id=name_index)
+        return term_rows, prefix_index, name_index
+
+    def encode_iri(self, iri_string: str, slot: Slot, statement: Statement) -> Rows:
+        """
+        Encode iri.
+
+        Args:
+            iri_string (str): full iri in string format.
+            slot (Slot): iri's place in statement.
+            statement (Statement): statement itself (Triple/Quad/GraphStart)
+
+        Returns:
+            Rows: extra rows for prefix and name tables, if any.
+
+        """
+        term_rows, prefix_index, name_index = self.encode_iri_indices(iri_string)
+        if slot == Slot.subject:
+            iri = statement.s_iri
+        elif slot == Slot.predicate:
+            iri = statement.p_iri
+        elif slot == Slot.object:
+            iri = statement.o_iri
+        elif slot == Slot.graph and isinstance(
```

Comment:
These extra isinstance checks are probably expensive, it would be good to avoid them.